### PR TITLE
Improvement #1056 TooGenericExceptionCaught

### PR DESCRIPTION
**Pull request by schalkms**
reference #1056

### CODE REVIEW

File: detekt-cli/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -188,6 +188,7 @@ exceptions:
      - IndexOutOfBoundsException
      - RuntimeException
      - Throwable
+    allowedExceptionNameRegex: "^(_|(ignore|expected).*)"
```
Comment (vanniktech) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: also allow `ignored`?

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt [hunk 2]
```diff
@@ -57,17 +61,28 @@ class TooGenericExceptionCaught(config: Config) : Rule(config) {
 	private val exceptions: Set<String> = valueOrDefault(
 			CAUGHT_EXCEPTIONS_PROPERTY, caughtExceptionDefaults).toHashSet()
 
+	private val allowedExceptionNameRegex = Regex(
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt]: Let's extract this regex and reuse between multiple rules?

File: detekt-cli/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (schalkms) [reply detekt-cli/src/main/resources/default-detekt-config.yml]: ignore`.*` is allowed 🔢

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt (hunk 2 above)
Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt]: will do

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt [hunk 3]
```diff
@@ -82,6 +83,15 @@ abstract class BaseConfig : Config {
 		is Boolean -> result.toBoolean()
 		is Double -> result.toDouble()
 		is String -> result
+		is List<*> -> parseList(result)
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt]: @arturbosch I hope we can live with this change. YML List couldn't be tested without this change.

File: detekt-cli/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (vanniktech) [reply detekt-cli/src/main/resources/default-detekt-config.yml]: You got me!

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt [hunk 4]
```diff
@@ -39,7 +43,8 @@ import org.jetbrains.kotlin.psi.KtCatchClause
  *			 - IndexOutOfBoundsException
  *			 - RuntimeException
  *			 - Throwable)
- *
+ * @configuration allowedExceptionNameRegex - ignores too generic exception types which match this regex
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/TooGenericExceptionCaught.kt]: Do we need this here? Seems like it's enough if it's in the generated README files.

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt (hunk 3 above)
Comment (arturbosch) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt]: I moved the list parse logic to `TestConfig` and merged this manually :)

**Pull request closed without merging**